core/src/main/java/org/elasticsearch/action/bulk/Retry.java

```diff
@@ -102,12 +125,15 @@ public BulkResponse withSyncBackoff(Client client, BulkRequest bulkRequest) thro
         private volatile BulkRequest currentBulkRequest;
         private volatile ScheduledFuture<?> scheduledRequestFuture;
 
-        AbstractRetryHandler(Class<? extends Throwable> retryOnThrowable, BackoffPolicy backoffPolicy, Client client, ActionListener<BulkResponse> listener) {
+        RetryHandler(Class<? extends Throwable> retryOnThrowable, BackoffPolicy backoffPolicy,
+                     BiConsumer<BulkRequest, ActionListener<BulkResponse>> consumer, ActionListener<BulkResponse> listener,
+                     Settings settings, BiFunction<TimeValue, Runnable, ScheduledFuture<?>> scheduleFn) {
```

Comment:
Ideally here you would just need to replace Client with ThreadPool, at least if we can get rid of Settings. Most of these changes should not be needed.